### PR TITLE
Y24-075 - Adds squix printers to PrinterList

### LINKED ISSUE #387: Y24-075 Add Squix printers
In Sample Prep team our old printers no longer work therefore we would like printers AA312bc and AA312bc2 added to Quanthub

Users
Lea lt12
Lynda ld20

Other
add to [PrinterList.json](https://github.com/sanger/quanthub/blob/e451bab5cda1f81bb7316cb557d503899b4dca28/src/config/PrinterList.json)

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #387 

#### Changes proposed in this pull request

- Adds printers `AA312bc` and `AA312bc2` to printer list
- Add `stub`, `morgan-plate-barcode` to printer list as non-production printers
- Refactor PrintJob.vue
- Migrate from Print My Barcode API from V1 to V2
- Update environment variables for V2 API
- Improve legability of message banners
- Improve layout of barcode print page
- Update readme
- Fix duplicated CSS build file
- Add build and test directories to .gitignore

### TODO

- [x] Complete response and error handling for createPrintJob in api/PrintMyBarcode.
- [ ] Design squix template/label for new printers. This will require input from story requestors as to what this label should look like. This will also require a new template in the deployment project and a pmb deployment.
- [x] Set correct fields for squix labels based off template in createLabels in api/PrintMyBarcode.
- [x] Unit tests for api/PrintMyBarcode. 
- [x] PrintJob cypress test.
- [x] Investigate `--slient` flag in "test:unit" command in package.json. I believe this is hiding some errors that we shouldn't be ignoring. Completed in #405

### CODE REVIEW

**Review by stevieing — COMMENTED** [review]
Quanthub currently uses print my barcode v1 so this won't work.

File: .env [hunk 1]
```diff
@@ -1,5 +1,5 @@
 NODE_ENV=development
 VITE_SEQUENCESCAPE_BASE_URL=http://localhost:3000/api/v2
 VITE_SEQUENCESCAPE_API_KEY=development
-VITE_PRINT_MY_BARCODE_BASE_URL=http://localhost:3000
-VITE_LABEL_TEMPLATE_ID=33
+VITE_PRINT_MY_BARCODE_BASE_URL=http://localhost:9292
+VITE_LABEL_TEMPLATE_NAME=sqsc_96plate_label_template_code128
```
Comment (stevieing) [inline .env:5]: Do we need this aynmore? The label template name is the same across instances.

**Review by stevieing — CHANGES_REQUESTED** [review]
Just one small thing.

It may also be worth removing server.js and json-server dependency.

Also worth creating a story to move the sequencescape request to fetch and then we can remove Axios (No need to do it in this story).

File: .env (hunk 1 above)
Comment (BenTopping) [reply .env]: Yeah good point, I will move it to the code directly.

**Comment by BenTopping** [comment]
I have removed jsorm dependency which we are no longer using but it had a peer dependency with lodash-es which we are using so i have added lodash-es explicitly.

**Comment by stevieing** [comment]
> I have removed jsorm dependency which we are no longer using but it had a peer dependency with lodash-es which we are using so i have added lodash-es explicitly.

We are using lodash in 1 place to do a groupBy. I might do a pull request to remove json server and it's usage. It seems overkill to use it for a single function.

**Comment by stevieing** [comment]
> > I have removed jsorm dependency which we are no longer using but it had a peer dependency with lodash-es which we are using so i have added lodash-es explicitly.
> 
> We are using lodash in 1 place to do a groupBy. I might do a pull request to remove json server and it's usage. It seems overkill to use it for a single function.

We are also importing the whole library into a component.

**Pull request merged**